### ISSUE TITLE
Test Cloud iOS: error CS0227: Unsafe code may only appear if compiling with /unsafe

### ISSUE THREAD

**Issue by jamesatgithub** [Microsoft/vsts-tasks#6067]
## Environment
- Server - VSTS 
      - If using VSTS, provide the account name, team project name, build definition name/build number: 
boundlss/Boundlss/iOS.build+uitest/Build 20171218.1

- Agent - Private:     
   - If using private agent, provide the OS of the machine running the agent and the agent version: latest XCode 9.2 and Xamarin.iOS 11.6

## Issue Description

the Xamarin.iOS UITest task (Build solution AppBoundlss.IOSTest/AppBoundlss.IOSTest
) failed with error like: error CS0227: Unsafe code may only appear if compiling with /unsafe

this error came from a Xamarin.iOS Binding project, but the Xamarin.iOS also references to this binding project and the Xamarin.iOS build task was build ok.

### Error logs

2017-12-18T02:07:36.1807520Z   obj/Release_Boundlss_Test/ios/ObjCRuntime/Trampolines.g.cs(123,23): error CS0227: Unsafe code may only appear if compiling with /unsafe [/Users/vso113581/vsts-agent/_work/1/s/AuthBinding/AuthBinding.csproj]

**Comment by lkillgore** [comment]
Hi @jamesatgithub, are you sure that the error message is coming from the Xamarin UITest task?  That message appears to be coming from the MSBuild task.  Something you may want to try: upgrading to a newer version of NuGet (perhaps something like 4.5?).

**Comment by TingtingAn** [comment]
@lkillgore    please refer to this link  https://gyazo.com/0bf2b772751777cd7633008cb21d387a 

it works for xamarin.ios build, but not for xamarin.uitest

**Comment by lkillgore** [comment]
Hi @TingtingAn, thanks for your reply.  I apologize, but I'm still not sure I understand your situation.  The error message in your log looks to me like it's coming from the MSBuild task and not the Xamarin.iOS task or the Xamarin Test Cloud task.  When you edit your build definition, which is the task that's failing?

I'm guessing it's one of these 3:
![image](https://user-images.githubusercontent.com/26548919/35224913-be89a642-ff54-11e7-990c-3002bcac74f6.png)

Also, when you say that it works for Xamarin.iOS but not for Xamarin.UITest, are you saying that one of the tasks is succeeding and the other task is failing?

Thanks!

**Comment by TingtingAn** [comment]
@lkillgore   Yes, it is a MSBuild task.  Please refer to this link: https://gyazo.com/31030825ec30e05adea03f4c7471de57 

In our solution, there is a xamarin.ios binding project. It is referenced by my Xamarin.ios project.
When build xamarin.ios project, it works well.  But it not work for xamarn.uitest build.   
It gave us unsafe code issue.

**Comment by lkillgore** [comment]
Hi @TingtingAn, thanks for your reply.  Is there a reason why you're not building the uitest as part of the solution?  Have you tried using the Xamarin.iOS task to build the uitest (instead of MSBuild)?

Something that may help you out is to try running the command line that's run with '[command]' as part of the MSBuild step.  It may be able to reproduce the issue on your developer box (or on your private build agent).

Also, from what I can tell, adding '/unsafe' as a parameter in the MSBuild task could very possibly fix your compile errors, though I'm unclear on what all the ramifications are for doing that.

**Comment by TingtingAn** [comment]
@lkillgore  mmm, I cannot use a solution for MSBuild, because that xamarin.ios build need specify a certicate.   And msbuild works well before for uitest.  After added a xamarin.ios binding project, it gave us the error.

**Comment by lkillgore** [comment]
Hi @TingtingAn, could you please explain what you mean when you say 'msbuild works well before for uitest'?

Also, did you try my other suggestions?  Could you explain their results?  Thanks!

**Comment by madhurig** [comment]
@jamesatgithub @TingtingAn : Were you able to resolve your issue?
